processing/src/main/java/org/apache/druid/query/operator/WindowOperatorQueryQueryToolChest.java

```diff
@@ -116,6 +128,36 @@ public Sequence<Object[]> resultsAsArrays(
     return (Sequence) resultSequence;
   }
 
+  @Override
+  public Optional<Sequence<FrameSignaturePair>> resultsAsFrames(
+      WindowOperatorQuery query,
+      Sequence<RowsAndColumns> resultSequence,
+      MemoryAllocatorFactory memoryAllocatorFactory,
+      boolean useNestedForUnknownTypes
+  )
+  {
+    RowSignature rowSignature = resultArraySignature(query);
+    RowSignature modifiedRowSignature = useNestedForUnknownTypes
+                                        ? FrameWriterUtils.replaceUnknownTypesWithNestedColumns(rowSignature)
```

Comment:
I don't understand why this is need ; what it does...

I think it would be very usefull to explain in the function's apidoc why `replaceUnknownTypesWithNestedColumns` could be needed - and why that's beneficial